server/src/main/java/com/thoughtworks/go/server/service/GoConfigService.java

```diff
@@ -1191,4 +1237,33 @@ public ConfigElementImplementationRegistry getRegistry() {
         return registry;
     }
 
+    public Map<String, Map> artifactIdToPluginIdForFetchPluggableArtifact(String stagePatent,
+                                                                          String currentPipelineName,
+                                                                          String currentStageName) {
+
+        final List<PipelineConfig> pipelineConfigs = stagePatent.equals("templates") ? getAllPipelineConfigs() : pipelinesForFetchArtifacts(currentPipelineName);
```

Comment:
Just a question: If, for the current pipeline, the dependency materials are not specified, then this returns empty, right? However, until the user clicks on save of fetch task, the validation about the pipeline not being a dependency material won't happen. Which means that, upon entering the artifactId in the fetch view, no plugin view shows up. At the time, we show the plugin dropdown, is it?

Reply:
Yes, the user will not get autosuggest unless they reload the page.